crates/libsolv_rs/src/pool.rs

```diff
@@ -0,0 +1,278 @@
+use crate::conda_util;
+use crate::id::{MatchSpecId, RepoId, SolvableId, StringId};
+use crate::solvable::{PackageSolvable, Solvable};
+use rattler_conda_types::{MatchSpec, PackageRecord};
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::str::FromStr;
+
+pub struct Pool<'a> {
+    pub(crate) solvables: Vec<Solvable<'a>>,
+
+    /// The total amount of registered repos
+    total_repos: u32,
+
+    /// Interned strings
+    pub(crate) strings_to_ids: HashMap<String, StringId>,
```

Comment:
I'd rather postpone this until we have benchmarks (it is kind of impossible that performance would worsen, but this might be a good way of forcing us to write them 😛 )